include/seastar/core/abort_source.hh

```diff
@@ -21,6 +21,10 @@
 
 #pragma once
 
+#if __cplusplus > 201703L
+#include <concepts>
+#endif
```

Comment:
Shouldn't this rather come right in `seastar/util/concepts.hh` ?

Reply:
I thought this header should only provide the macro definition, not the header from standard library. but both you and Xuehan are proposing this change, I am not quite sure now. 😅

Reply:
If the `#if __cplusplus > 201703L` wasn't there I wouldn't suggest it :laughing:  But I've a feeling that this construct will follow us pretty much everywhere (until we stop supporting "ancient" enough compilers/standards)

Reply:
Yes, concepts.hh is a stand-in for `<concepts>` so we don't need this conditional everywhere.

Reply:
also, there is __has_include and feature test macros, better than checking __cplusplus.

Reply:
created #1458 as a fall back. but i'd prefer using #1445 if @xxhdx1985126 could update his PR.